docs/source/en/tasks/semantic_segmentation.md

```diff
@@ -689,7 +689,8 @@ Reload the dataset and load an image for inference.
 We will now see how to infer without a pipeline. Process the image with an image processor and place the `pixel_values` on a GPU:
 
 ```py
->>> device = torch.device("cuda" if torch.cuda.is_available() else "cpu")  # use GPU if available, otherwise use a CPU
+>>> from accelerate.test_utils.testing import get_backend
+>>> device, _, _ = get_backend()
```

Comment:
Maybe add a comment here that explains what's happening. Perhaps something like:

```suggestion
# automatically detects the underlying device type (GPU, CUDA, CPU, XPU, MPS, etc.)
>>> device, _, _ = get_backend()
```